external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -304,6 +304,14 @@ pub struct ParsingSymbolicator<'a> {
     file_id_mapping: &'a HashMap<FileHash, usize>,
     // A mapping from file IDs to a split vector of the lines in each file (used to build docstrings)
     file_id_to_lines: &'a HashMap<usize, Vec<String>>,
+    /// Associates uses for a given definition to allow displaying all references
+    references: &'a mut BTreeMap<DefLoc, BTreeSet<UseLoc>>,
+    /// A UseDefMpa for a given module (needs to be appropriately set before the module
+    /// processing starts)
+    use_defs: UseDefMap,
+    /// Module name lengths in access paths for a given module (needs to be appropriately
+    /// set before the module processing starts)
+    mod_name_lengths: BTreeMap<Position, usize>,
```

Comment:
I did not want to thread `mod_name_lengths` through all the symbolicator calls (for both symbolicators) in the same style that `references` and `use_defs` were threaded before, so I made it part of the symbolicator and refactored the code to have `references` and `use_defs` follow suit even though this refactoring was not strictly necessary to implement the new feature. Hopefully it ultimately made the code cleaner, though.